_posts/2022-01-28-filesystem-encryption-integrity.adoc

```diff
@@ -0,0 +1,145 @@
+---
+layout: post
+title: "Encrypting Filesystem realms and enabling integrity"
+date: 2022-01-28
+tags: filesystem-realm encryption integrity
+synopsis: An overview of the new filesystem realm encryption and integrity in the Elytron Security Realms.
+author: araskar
+link:
+---
+
+Previously, any data in the filesystem realm was always stored unencrypted and data was not verified. This meant that anyone with access to the filesystem identity files could access the credentials, and attributes (such as assigned roles). Along with that if anyone were to change the data in file, the realm would have no way of knowing. Now both encryption has been added via a SecretKey, and filesystem integrity has been added by generating a signature via a Public and Private Keypair.
```

Comment:
s/change the data/tamper with the data

Reply:
s/via a Public and Private Keypair/via a key pair

Reply:
Changed